internal/controller/controllers/spoke_k8s_client.go

```diff
@@ -109,9 +111,60 @@ func (c *spokeK8sClient) ApproveCsr(csr *certificatesv1.CertificateSigningReques
 }
 
 func (c *spokeK8sClient) GetNode(name string) (*corev1.Node, error) {
-	node, err := c.nodesClient.Get(context.TODO(), name, metav1.GetOptions{})
+	node := &corev1.Node{}
+	if err := c.Get(context.TODO(), types.NamespacedName{Name: name}, node); err != nil {
+		return nil, err
+	}
+	return node, nil
+}
+
+func (c *spokeK8sClient) StartAgentForReclaim(ctx context.Context, agent *aiv1beta1.Agent, image string) error {
+	hostname := getAgentHostname(agent)
```

Comment:
Maybe we should consider leaving this client to be more generic, and just add a function CreatePod, or CreatePosOnHost ?

Reply:
:man_shrugging:

The generic client is `client.Client`.
Honestly `GetNode`, and `ListCsrs` feel like they shouldn't exist. Those functions don't add anything meaningful over `client.Client`.
I'd rather see this client object wrap some significant logic that maybe we don't want to include in other files.

`ApproveCsr` is a pretty good example. We're adding a message that it was approved by assisted service and we need to build a condition in a specific way. I actually think it could include even more logic if it were up to me.

Is there anything in particular you object to in here? As far as I can tell the hostname is all I really need from the agent (this is still WIP after all), so maybe if I just passed the hostname as a parameter?

Reply:
Actually in the case of `ListCsrs` I think it _definitely_ shouldn't exist. It's just an alias that prevents us from using contexts properly.